src/transformers/pipelines/conversational.py

```diff
@@ -232,6 +233,10 @@ class ConversationalPipeline(Pipeline):
     """
 
     def __init__(self, *args, **kwargs):
+        warnings.warn(
+            "`ConversationalPipeline` is now deprecated, and the functionality has been moved to the standard `text-generation` pipeline, which now accepts lists of message dicts as well as strings. This class will be removed in a future release.",
```

Comment:
I'd add a specific version in which the class will be deprecated (v4.40, i.e. two minor releases after next minor release?)

Reply:
Done! I picked 4.42 to give people more time.